docs/raw/push-notification-server.md

```diff
@@ -0,0 +1,672 @@
+---
+permalink: /spec/16
+parent: Stable specs
+title: 16/PUSH-NOTIFICATION-SERVER
+---
+
+# 16/PUSH-NOTIFICATION-SERVER
+
+> Version: 0.1
+>
+> Status: Raw
+>
+> Authors: Andrea Maria Piana <andreap@status.im>
+
+- [Push Notification Server](#16push-notification-server)
+  - [Reason](#reason)
+  - [Requirements](#requirements)
+  - [Registering with the push notification service](#registering-with-the-push-notification-server)
+  - [Re-registering with the push notification service](#re-registering-with-the-push-notification-server)
+  - [Changing options](#changing-options)
+  - [Unregistering from push notifications](#unregistering-from-push-notifications)
+  - [Advertising a push notification server](#advertising-a-push-notification-server)
+  - [Discovering a push notification server](#discovering-a-push-notificatino-server)
+  - [Querying the push notification service](#querying-the-push-notification-server)
+  - [Sending a push notification](#sending-a-push-notification)
+  - [Flow](#flow)
+    - [Registration process](#registration-process)
+    - [Sending a notification](#sending-a-notification)
+    - [Receiving a notification](#receiving-a-notification)
+  - [Protobuf description](#protobuf-description)
+    - [PushNotificationRegister](#pushnotificationregister)
+    - [PushNotificationPreferences](#pushnotificationpreferences)
+    - [PushNotificationDeviceToken](#pushnotificationdevicetoken)
+    - [PushNotificationOptions](#pushnotificationoptions)
+    - [PushNotificationPreferencesAcknowledgment](#pushnotificationpreferencesacknowledgement)
+    - [ContactCodeAdvertisement](#contactcodeadvertisement)
+    - [PushNotificationAdvertisementInfo](#pushnotificationadvertisementinfo)
+    - [PushNotificationQuery](#pushnotificationquery)
+    - [PushNotificationQueryInfo](#pushnotificationqueryinfo)
+    - [PushNotificationQueryResponse](#pushnotificationqueryresponse)
+    - [PushNotification](#pushnotification)
+    - [PushNotificationRequest](#pushnotificationrequest)
+    - [PushNotificationAcknowledgement](#pushnotificationacknowledgement)
+  - [Partitioned topic](#partitioned-topic)
+  - [FAQ](#faq)
+  - [Changelog](#changelog)
+    - [Version 0.1](#version-01)
+  - [Copyright](#copyright)
+
+## Reason
+
+Push notifications for IOS devices and for some android device can only be
```

Comment:
Comment for second round: is this strictly true? in practice this is the conclusion. I'd perhaps elaborate a bit more remote and local options here and why remote for iOS makes sense, and that this implies using the APN service

Reply:
well depends on your definition of push notification, if you want them to be instant (which is the common understanding), then yes, as we are not allowed to keep a connection open (only voip applications do), and the app would be brought to the foreground to check for messages at intervals, and only when apple things is energy efficient (on wifi, when the user switches application etc), and would not be brought back if force quit.

I can add a section to expand on the rationale.

Reply:
I agree with the conclusion, just that the decision space pruning should come first, i.e. can only do remote or local, local does'nt work due to X. If X were were to change in future (unlikely but hey), it is clear to reader that architecture could change.